src/Core/Resolvers/QueryExecutor.cs

```diff
@@ -805,5 +853,24 @@ private void ValidateSize(long availableSizeBytes, long sizeToBeReadBytes)
                     subStatusCode: DataApiBuilderException.SubStatusCodes.ErrorProcessingData);
             }
         }
+
+        internal virtual void AddDbExecutionTimeToMiddlewareContext(long time)
+        {
+            HttpContext? httpContext = HttpContextAccessor?.HttpContext;
+            if (httpContext != null)
+            {
+                // locking is because we could have multiple queries in a single http request and each query will pr processed in parallel leading to concurrent access of the httpContext.Items.
+                lock (_httpContextLock)
+                {
+                    if (!httpContext.Items.ContainsKey(TOTALDBEXECUTIONTIME))
+                    {
+                        httpContext.Items[TOTALDBEXECUTIONTIME] = (long)0;
+                    }
+
+                    long currentTime = (long)httpContext.Items[TOTALDBEXECUTIONTIME]!;
+                    httpContext.Items[TOTALDBEXECUTIONTIME] = currentTime + time;
```

Comment:
Can this be simplified so that there is only 1 read and 1 write? (e.g. TryGetValue() https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.idictionary-2.trygetvalue?view=net-8.0)
-> if key doesn't exist (First instance of reading httpContext.Items), then you can just write to `Items` a single time. Avoiding the second read/write.
-> if key exists -> no need to use null forgiving operator.

Reply:
Fixed.